### DISCUSSION TITLE
How to use composer install with "--with-all-dependencies" option

### DISCUSSION THREAD

**Discussion by felixranesberger** [renovatebot/renovate#22132] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to update all `typo3/*` dependencies with the `--with-all-dependencies` composer option.

### What you tried so far.

I've found this GitHub issue, but I'm still not sure how to fix the issue:
https://github.com/renovatebot/renovate/issues/6295

### Relevant debug logs

# Renovate Project Configuration

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    ":dependencyDashboard",
    ":semanticPrefixFixDepsChoreOthers",
    ":ignoreModulesAndTests",
    "group:monorepos",
    "group:recommended",
    "workarounds:all"
  ],
  "packageRules": [
    {
      "groupName": "Stylelint",
      "matchPackagePatterns": [
        "stylelint"
      ]
    },
    {
      "groupName": "ESLint",
      "matchPackagePatterns": [
        "eslint"
      ]
    },
    {
      "groupName": "Vite",
      "matchPackagePatterns": [
        "vite"
      ]
    },
    {
      "groupName": "TYPO3",
      "matchPackagePrefixes": [
        "typo3/"
      ],
      "assignees": ["@username"]
    },
    {
      "matchPackagePatterns": [
        "^typo3/"
      ],
      "separateMinorPatch": true
    },
    {
      "matchPackagePatterns": [
        "^typo3/"
      ],
      "matchUpdateTypes": [
        "minor",
        "major"
      ],
      "enabled": false
    }
  ]
}
```

**Comment by rarkins** [comment]
#6295 is open, which means unimplemented.

If you can control Composer with an env variable for the same thing, you could try `customEnvVariables`.

**Reply by felixranesberger** [reply]
Unfortunately this is not possible to my knowledge.

I would like to fix this issue, so I commented on the [pull request](https://github.com/renovatebot/renovate/issues/6295#issuecomment-1592940012) and asked for help 👍

**Comment by christophs78** [comment]
Just run into the same issue. (Want to do automated patch-updates for Typo3.)
When we look into Composer-docu (https://getcomposer.org/doc/03-cli.md#environment-variables) we see there´s no environment-variable for activating `--with-all-dependencies`.
We can do COMPOSER_IGNORE_PLATFORM_REQS via environment-variable but not this.
